scripts/production/homology_stats_rapid.py

```diff
@@ -0,0 +1,189 @@
+            taxid = dict_row["taxon_id"]
+            if taxid == query_taxid:
+                continue
+            anc = Taxonomy.all_common_ancestors(session, query_taxid, taxid)
+            res.append((dict_row["genome_db_id"], dict_row["name"], taxid, len(anc)))
+    res = sorted(res, key=lambda x: (x[3], x[2]), reverse=True)
+    return res[0][0], res[0][1], res[0][2], query_species, query_taxid
+
+
+def query_rr_database(
+    rr_dbc: Connection, genome_db_id: int, ref_db: str
+) -> Tuple[int, int]:
+    """
+    Query the database for the number of homologs and genes.
+
+    Args:
+        rr_dbc (Connection): Rapid release database connection
+        genome_db_id (int): Genome database ID
+        ref_db (str): Reference database name
+
+    Returns:
+        Tuple[int, int]: Tuple containing the number of homologs and genes
+    """
+    connection: Connection = rr_dbc.connect()
+
+    # Query nr_homologs
+    query1 = text(
+        f"""
+        SELECT COUNT(DISTINCT gene_member_id) AS nr_homologs
+        FROM homology
+        JOIN homology_member USING(homology_id)
+        JOIN {ref_db}.gene_member gm USING(gene_member_id)
+        JOIN {ref_db}.genome_db gdb USING(genome_db_id)
+        WHERE gm.biotype_group = 'coding'
+        AND homology.description LIKE 'homolog_%'
+        AND gdb.genome_db_id = {genome_db_id}
+    """
+    )
+    result1 = connection.execute(query1).fetchone()
+    nr_homologs = result1["nr_homologs"]
+
+    # Query nr_genes
+    query2 = text(
+        """
+        SELECT COUNT(gene_member_id) AS nr_genes
+        FROM gene_member
+        JOIN genome_db USING(genome_db_id)
+        WHERE gene_member.biotype_group = 'coding'
+        AND genome_db_id = 1
+    """
+    )
+    result2 = connection.execute(query2).fetchone()
+    nr_genes = result2["nr_genes"]
+
+    return nr_homologs, nr_genes
+
+
+def write_results_to_json(output_file: str, data: dict) -> None:
+    """
+    Write results to a JSON file.
+
+    Args:
+        output_file (str): Output file name
+        data (Dict): Data to write to JSON
+    """
+    with open(output_file, "w", encoding="utf-8") as outfile:
+        json.dump(data, outfile)
+
+
+def main() -> None:
+    """
+    Main function to run the script.
+    """
+    args = parse_arguments()
+    db_url = args.database
+    output_file = args.output
+    ref_db = args.ref_db
+
+    tax_dbc = DBConnection(db_url)
+    rr_dbc = create_engine(db_url, future=True)
+
+    ref_gdb, ref_species, ref_taxid, query_species, query_taxid = get_closest_ref(
+        tax_dbc, rr_dbc, ref_db
+    )
+
+    nr_homologs, nr_genes = query_rr_database(rr_dbc, ref_gdb, ref_db)
+    json_data = {
+        "query_species": query_species,
+        "query_taxon_id": query_taxid,
+        "ref_species": ref_species,
+        "ref_taxon_id": ref_taxid,
+        "nr_query_genes": nr_genes,
+        "nr_homologies": nr_homologs,
+    }
+    write_results_to_json(output_file, json_data)
```

Comment:
If `write_results_to_json` is to be used only in this instance, I'd suggest to use its code directly here: 
```suggestion
    with open(args.output, "w", encoding="utf-8") as outfile:
        json.dump(json_data, outfile)
```
Kudos for using `UTF-8`. 😃